src/doc/rustdoc/src/how-to-write-documentation.md

```diff
@@ -0,0 +1,78 @@
+# How to write documentation
+
+This chapter intends to explain how to not only write documentation but also on
+how to write **good** ones. But something to keep in mind when writing
+documentation: you write as much for others as for you so it's important to be
+clear! The more the better. If an item is public, then it should be documented.
+
+## Basic structure
+
+If possible, each item's documentation should try to follow this structure:
+
+```text
+[short sentence explaining what it is]
+
+[more detailed explanation]
+
+[at least one code example that users can copy/paste to try it]
+
+[even more advanced explanations if necessary]
+```
+
+Pretty simple but important to follow. The code example is really important:
+it might help your users to have a better understanding of what an item is
+used for and/or how to use it.
+
+Let's see an example coming from the [standard library] by taking a look at the
+[env::args] function:
+
+``````text
+Returns the arguments which this program was started with (normally passed
+via the command line).
+
+The first element is traditionally the path of the executable, but it can be
+set to arbitrary text, and may not even exist. This means this property should
+not be relied upon for security purposes.
+
+On Unix systems shell usually expands unquoted arguments with glob patterns
+(such as `*` and `?`). On Windows this is not done, and such arguments are
+passed as-is.
+
+# Panics
+
+The returned iterator will panic during iteration if any argument to the
+process is not valid unicode. If this is not desired,
+use the [`args_os`] function instead.
+
+# Examples
+
+```
+use std::env;
+
+// Prints each argument on a separate line
+for argument in env::args() {
+    println!("{}", argument);
+}
+```
+
+[`args_os`]: ./fn.args_os.html
+``````
+
+As you can see, it follows the structure detailed above: it starts with a short
+sentence explaining what the functions does, then it provides more information
+and finally provides a code example.
+
+## Markdown
+
+The documentation is using the [commonmark markdown specification]. You might be
+interested into taking a look at their website to see what's possible to do!
```

Comment:
Here I think it's important to make it clear that `rustdoc` uses commonmark, and if possible some of the text about taking a look should try and link to the requisite part of the commonmark site.  I also dislike the `!` :smile: